### DISCUSSION TITLE
Conditional syntax for custom fields

### DISCUSSION THREAD

**Discussion by lucasssfigueredo** [rfennell/AzurePipelines#970] [Q&A] [answered]
Hi.

I created a custom field and would like to filter the work items by it. Can you help me with the If syntax?

I tried the following, but got an error.

{{#forEach workItems}} 

{{#if {{lookup this.fields 'Custom.AddtoReleaseNotes'}} == true}}

##{{this.id}} {{lookup this.fields 'System.Title'}}

Type: {{lookup this.fields 'System.WorkItemType'}}

{{{lookup this.fields 'Custom.AddtoReleaseNotes'}}}
{{{lookup this.fields 'Custom.Releasenotes'}}}

{{/if}}
{{/forEach}}

**Comment by rfennell** [comment, marked as answer]
I don't think there is an out the box string comparator in Handlebars [see this post](https://stackoverflow.com/questions/34252817/handlebarsjs-check-if-a-string-is-equal-to-a-value)

So, off the top of my head, you can use something like the following using one of the [handlebar helpers](https://github.com/helpers/handlebars-helpers)

```
{{#forEach this.workItems}}
{{#if isFirst}}### Only show this line on the first time around the loop - a title{{/if}}
{{#if (contains (lookup this.fields ''Custom.AddtoReleaseNotes'') 'true')}}
##{{this.id}} {{lookup this.fields 'System.Title'}}
Type: {{lookup this.fields 'System.WorkItemType'}}
{{{lookup this.fields 'Custom.AddtoReleaseNotes'}}}
{{{lookup this.fields 'Custom.Releasenotes'}}}
{{/if}}
{{/forEach}}
````

This is based on a past issue #710 and I know it works (though that was comparing tag fields, it is the same ide)

I am sure you could swap the 

`{{#if (contains (lookup this.fields ''Custom.AddtoReleaseNotes'') 'true')}}`

for

`{{#if (eq (lookup this.fields ''Custom.AddtoReleaseNotes'') 'true')}}`

But I have not tested that

**Reply by lucasssfigueredo** [reply]
Thanks a lot Richard.

The second option worked fine:

{{#if (eq (lookup this.fields 'Custom.AddtoReleaseNotes') true)}}
